crates/vhost/src/vhost_user/slave_req_handler.rs

```diff
@@ -609,11 +609,10 @@ impl<S: VhostUserSlaveReqHandler> SlaveReqHandler<S> {
         if size - mem::size_of::<VhostUserConfig>() != msg.size as usize {
             return Err(Error::InvalidMessage);
         }
-        let flags: VhostUserConfigFlags;
-        match VhostUserConfigFlags::from_bits(msg.flags) {
-            Some(val) => flags = val,
+        let flags: VhostUserConfigFlags = match VhostUserConfigFlags::from_bits(msg.flags) {
+            Some(val) => val,
             None => return Err(Error::InvalidMessage),
-        }
+        };
```

Comment:
This change seems unrelated (I prefer your cleanup), maybe you can do it in a separate commit.

Reply:
I think you can use `ok_or` here.

Reply:
I changed it because I saw CI failed due to clippy, but I think that may actually have been for a different reason, so I can probably drop this / make a separate PR to make clippy happy.

Reply:
Sorry, I misspoke!
The fix is fine, especially if there is a new clippy warning (I guess with a new version of Rust), I was just asking to do it in a separate commit, even within this PR :-)

Reply:
Done. :)